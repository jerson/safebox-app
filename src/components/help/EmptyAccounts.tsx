import React from 'react';
import {
  StyleSheet,
  ViewStyle,
  TextStyle,
  View,
  ImageStyle,
  Image
} from 'react-native';
import Colors from '../../modules/constants/Colors';
import { useNavigation } from 'react-navigation-hooks';
import Button from '../ui/Button';
import Text from '../ui/Text';

const styles = StyleSheet.create({
  logo: {
    width: 90,
    height: 90
  } as ImageStyle,
  logoContent: {
    backgroundColor: Colors.grey6,
    borderRadius: 22,
    borderWidth: 1,
    borderColor: Colors.accent,
    alignSelf: 'center',
    overflow: 'hidden',
    marginBottom: 10
  } as ViewStyle,
  container: {
    alignItems: 'center',
    justifyContent: 'center',
    padding: 30
  } as ViewStyle,
  button: {
    marginTop: 20
  } as ViewStyle,
  description: {
    textAlign: 'center',
    color: Colors.grey5
  } as TextStyle
});

function EmptyAccounts() {
  const { navigate } = useNavigation();
  const goToAdd = () => {
    navigate('AddAccount');
  };
  return (
    <View style={styles.container}>
      <View style={styles.logoContent}>
        <Image
          style={styles.logo}
          resizeMode={'stretch'}
          source={require('../../assets/images/logo.png')}
        />
      </View>

      <Text style={styles.description} weight={'Light'}>
        {'The first thing you should do is add a '}
        <Text weight={'SemiBold'}>{'Secret account'}</Text>
      </Text>

      <Button
        icon={'plus-circle'}
        style={styles.button}
        title={'Add first secret account'}
        typeColor={'primaryLight'}
        onPress={goToAdd}
      />
    </View>
  );
}

export default EmptyAccounts;
