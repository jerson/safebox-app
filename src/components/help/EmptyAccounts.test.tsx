import React from 'react';
import * as renderer from 'react-test-renderer';
import EmptyAccounts from './EmptyAccounts';

const props = {};
test('renders without crashing', () => {
  const snapshot = renderer.create(<EmptyAccounts {...props} />).toJSON();
  expect(snapshot).toBeTruthy();
});

test('render just views', () => {
  const snapshot = renderer.create(<EmptyAccounts {...props} />).toJSON();
  expect(snapshot).toMatchSnapshot();
});
