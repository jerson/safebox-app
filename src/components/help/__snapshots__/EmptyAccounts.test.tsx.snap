// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render just views 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "padding": 30,
    }
  }
>
  <View
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#38414e",
        "borderColor": "#c2d0e7",
        "borderRadius": 22,
        "borderWidth": 1,
        "marginBottom": 10,
        "overflow": "hidden",
      }
    }
  >
    <Image
      resizeMode="stretch"
      source={
        Object {
          "testUri": "../../../src/assets/images/logo.png",
        }
      }
      style={
        Object {
          "height": 90,
          "width": 90,
        }
      }
    />
  </View>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Nunito",
          "fontSize": undefined,
          "fontStyle": "normal",
          "fontWeight": "300",
        },
        Object {},
        Object {
          "color": "#65748b",
          "textAlign": "center",
        },
      ]
    }
  >
    The first thing you should do is add a 
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Nunito",
            "fontSize": undefined,
            "fontStyle": "normal",
            "fontWeight": "600",
          },
          Object {},
          undefined,
        ]
      }
    >
      Secret account
    </Text>
  </Text>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#9066f5",
        "borderColor": "#9066f5",
        "borderRadius": 10,
        "flexDirection": "row",
        "justifyContent": "center",
        "margin": 0,
        "marginTop": 20,
        "opacity": 1,
        "padding": 12,
        "zIndex": 1,
      }
    }
    testID="button_touchable"
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 14,
              "marginRight": 4,
              "top": 3,
            },
            Object {
              "color": "#FFFFFF",
            },
            undefined,
          ],
          Object {
            "fontFamily": "Feather",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      ï†µ
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Nunito",
            "fontSize": undefined,
            "fontStyle": "normal",
            "fontWeight": "400",
          },
          Object {},
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 14,
              "textAlign": "center",
            },
            Object {
              "color": "#FFFFFF",
            },
            undefined,
          ],
        ]
      }
    >
      Add first secret account
    </Text>
  </View>
</View>
`;
